File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        nums = []
        for x in range(len(strs)):
            nums.append([0]*26)
        dict = {}
        for i in range(len(strs)):
            for j in range(len(strs[i])):
                nums[i][ord(strs[i][j])-ord('a')]+=1
        answer=[]
        group=0
        for i in range(len(strs)):
            if tuple(nums[i]) in dict:
                answer[dict[tuple(nums[i])]].append(strs[i])
            else:
                dict[tuple(nums[i])]=group
                answer.append([])
                answer[group].append(strs[i])
                group+=1
        return answer

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n=nums.size();
        int lo=0,hi=n-1,mid;
        while(lo<=hi){
            if(hi==lo){
                return nums[hi]==target? hi: -1;
            }
            mid=(hi+lo)/2;
            if(nums[mid] < target){
                lo=mid+1;
            }
            else if(nums[mid] > target){
                hi=mid-1;
            }
            else return mid;
        }
        return -1;
    }
};
